wikipedia_sentiment.py

```diff
@@ -0,0 +1,103 @@
+#scrapes wikipedia and performs sentiment analysis to figure out what wikipedia thinks about
+#congress democrats and republicans
+
+#imports pattern web
+from pattern.web import *
+#imports pattern natural language
+from pattern.en import * 
+
+def sentiment_finder(searchterm):
+    """Finds sentiment of a given article on wikipedia.
+        accepts string searchterm where searchterm is the title of a wikipedia article.
+        Returns vector of (sentiment, objectivity)
+        where sentiment is between -1.0 and 1.0 and objectivity is between 0 and 1"""
+    try:
+        article = Wikipedia().search(searchterm) #pull article from wikipedia 
+        articletext = str.splitlines(article.plaintext().encode('utf-8')) # plain text as list of string lines of article
+        founde = False #looking for articles that start with bulleted lists
+        for i in range(len(articletext)): #loop thru and find the first line in the article 
+                if articletext[i] == '* e':   #First line of article content marked by *e if article starts with bulleted lists
+                        index = i
+                        founde = True
+                if founde == False:          #if we haven't found e yet, no bulleted list so we automatically pull 1st paragraph
+                        index = 0
+                
+        return sentiment(articletext[index+2])     #return sentiment. +2 accounts for first empty lines
+    except:
+                                      #surprisingly some congressmen don't have wikipedia articles.
+       return (0,0)                   #returns no wikipedia sentiment if error is thrown
+                                      #(wikipedia article does not exist)
+    
+#populate list of current voting members of house and senate     
+    
+republicans = [] #empty list for republican names
+
+democrats = [] #empty list for democrat names
+
+
+senators = Wikipedia().search('Current members of the United States House of Representatives')  #pulls wikipedia article for senators
+senatorsections = senators.sections[5] #pulls section of senators wikipedia article to read (voting members)
+senatorlist = senatorsections.tables[0]#pulls list of current senators in table form
+
+#iterates thru the table and classifies article names with Democrats and Republicans
+for i in range(len(senatorlist.rows)):      
```

Comment:
This is totally fine, but there is this awesome thing called `enumerate()` that gives you both the current index and the values of the thing you are iterating through. I only recently started using it in my code and it's great. Learn more here, if interested: http://book.pythontips.com/en/latest/enumerate.html